api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -0,0 +1,95 @@
+package org.pmiops.workbench.db.model;
+
+import java.util.List;
+import javax.persistence.*;
+
+@Entity
+@Table(name = "institution")
+public class DbInstitution {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "institution_id")
+  private long institutionId; // primary opaque key for DB use only
+
+  @Column(name = "api_id", nullable = false, unique = true)
+  private String apiId; // unique key exposed to API
+
+  @Column(name = "long_name", nullable = false)
+  private String longName;
+
+  @Column(name = "organization_type_enum")
+  private Short organizationTypeEnum;
+
+  @Column(name = "organization_type_other_text")
+  private String organizationTypeOtherText;
+
+  @OneToMany(mappedBy = "institution", fetch = FetchType.LAZY)
+  private List<DbInstitutionEmailDomain> emailDomains;
+
+  @OneToMany(mappedBy = "institution", fetch = FetchType.LAZY)
+  private List<DbInstitutionEmailAddress> emailAddresses;
+
+  public DbInstitution() {}
+
+  public DbInstitution(final String apiId, final String longName) {
+    setApiId(apiId);
```

Comment:
I don't mind overloaded constructors in Java, and use them frequently. One thing to watch out for though, is if different versions of the constructor don't produce the same object when given consistent results.

For example, if I have a 1-arg constructor, and a 2-arg constructor, I generally expect the first one to call the second with `Constructor(myOnlyArg, DEFAULT_SECOND_ARG)` so that both give consistent results.

All that to say, I'd either drop your no-arg constructor, or have it call the 2-arg one with some reasonable default values. If defaults can't be chosen safely, then that's a smell you don't want a no-arg ctor.

Reply:
I'm pretty sure we don't have much choice here, see https://stackoverflow.com/questions/32467520/hibernate-and-no-arg-constructor/32467567 and https://docs.jboss.org/hibernate/core/3.5/reference/en/html/persistent-classes.html#persistent-classes-pojo-constructor